src/shared/fdset.c

```diff
@@ -158,6 +216,28 @@ int fdset_remove(FDSet *s, int fd) {
         return hashmap_remove(MAKE_HASHMAP(s), INT_TO_PTR(ret)) ? fd : -ENOENT;
 }
 
+int fdset_remove_indexed(FDSet *s, int index) {
+        _cleanup_close_ int fd = -EBADF;
+
+        assert(s);
+        assert(index >= 0);
+
+        /* Avoid integer overflow in FD_TO_PTR() */
+        if (index == INT_MAX)
+                return log_debug_errno(SYNTHETIC_ERRNO(ENOENT), "Refusing invalid index: %d", index);
```

Comment:
this check is unnecessary, as INT_MIN = -INT_MAX-1, hence still in the valid range.